src/main/java/org/breedinginsight/brapi/v2/BrAPIGermplasmController.java

```diff
@@ -105,19 +113,88 @@ public HttpResponse<Object> searchGermplasm(
         String extRefId = program.get().getId().toString();
         body.externalReferenceIds(List.of(extRefId));
 
+        // convert request filter dbIds from DeltaBreed UUID to BrAPI service dbIds for now
+        // TODO: all dbIds, just germplasmDbIds for now since that's what Field Book needs
+        // Could be more performant by doing batch lookup but at least just going to cache
+        List<String> convertedDbIds = new ArrayList<>();
+        for (String germplasmDbId : body.getGermplasmDbIds()) {
+            BrAPIGermplasm germplasm = germplasmService.getGermplasmByUUID(program.get().getId(), germplasmDbId);
+            convertedDbIds.add(germplasm.getGermplasmDbId());
+        }
+        body.setGermplasmDbIds(convertedDbIds);
+
         ApiResponse<Pair<Optional<BrAPIGermplasmListResponse>, Optional<BrAPIAcceptedSearchResponse>>> brapiGermplasm;
         brapiGermplasm = brAPIEndpointProvider
                 .get(programDAO.getCoreClient(program.get().getId()), GermplasmApi.class)
                 .searchGermplasmPost(body);
 
+        return getObjectHttpResponse(brapiGermplasm, program.get().getKey());
+    }
```

Comment:
```suggestion
        // convert request filter dbIds from DeltaBreed UUID to BrAPI service dbIds for now
        // TODO: all dbIds, just germplasmDbIds for now since that's what Field Book needs
        // Could be more performant by doing batch lookup but at least just going to cache
             body.setGermplasmDbIds(convertDbIdsToServiceIds(program.get().getId(), body.getGermplasmDbIds()));

        ApiResponse<Pair<Optional<BrAPIGermplasmListResponse>, Optional<BrAPIAcceptedSearchResponse>>> brapiGermplasm;
        brapiGermplasm = brAPIEndpointProvider
                .get(programDAO.getCoreClient(program.get().getId()), GermplasmApi.class)
                .searchGermplasmPost(body);

        return getObjectHttpResponse(brapiGermplasm, program.get().getKey());
    }

    private List<String> convertDbIdsToServiceIds(UUID programId, List<String> germplasmDbIds) {
        return germplasmDbIds.stream()
                .map(dbId -> {
                    try {
                        BrAPIGermplasm germplasm = germplasmService.getGermplasmByUUID(programId, dbId);
                        return germplasm.getGermplasmDbId();
                    } catch (Exception e) {
                        log.warn("Failed to convert dbId: " + dbId, e);
                        return null;
                    }
                })
                .filter(Objects::nonNull)
                .collect(Collectors.toList());
    }
```

Reply:
I'm good with breaking out a separate method but as far as the exception handling I think it'd be better to let that propagate up so we get an internal server error. I think it should be all or nothing and not tolerate any failure to convert ids.

Reply:
Ended up writing new service/DAO methods to convert all ids in one batch so could be done inline and a separate method is not longer needed.